internal/pkg_npm/pkg_npm.bzl

```diff
@@ -167,77 +173,102 @@ def create_package(ctx, deps_sources, nested_packages):
     """
 
     stamp = ctx.attr.node_context_data[NodeContextInfo].stamp
-    package_dir = ctx.actions.declare_directory(ctx.label.name)
-    package_path = ctx.label.package
 
-    # List of dependency sources which are local to the package that defines the current
-    # target. Also include files from external repositories that explicitly specified in
-    # the vendor_external list. We only want to package deps files which are inside of the
-    # current package unless explicitely specified.
-    filtered_deps_sources = _filter_out_external_files(ctx, deps_sources, package_path)
+    if not stamp and len(files) == 1 and files[0].is_directory and len(ctx.files.srcs) == 0 and len(ctx.files.nested_packages) == 0:
+        # Special case where these is a single dep that is a directory artifact and there are no
+        # source files or nested_packages; in that case we assume the package is contained within
+        # that single directory and there is no work to do
+        package_dir = files[0]
+    else:
```

Comment:
nit: could maybe use a short-circuit `return package_dir` to avoid needing extra indent on this very long else clause
(though I think since pack/publish scripts are needed in both cases you'd have to reorder things, or maybe extract a helper for packaging up the directory)

Reply:
👍